packages/engine-server/src/drivers/file/NoteParserV2.ts

```diff
@@ -264,7 +262,7 @@ export class NoteParserV2 {
     );
     await Promise.all(
       domains.map(async (domain) => {
-        return SchemaUtils.matchDomain(domain, notesById, schemas);
+        return SchemaUtils.matchDomain(domain, notesById, this.engine);
```

Comment:
This is more for efficiency, but do you think it's better to leave this as before and pass in the SchemaModuleDict instead?

`matchDomain` is only used in the note parser and when we call `NoteParserV2.parseFiles`, we already have the schema dict generated from the engine.

Otherwise, we'd have to call getSchema for every domain which ends up cloning each schema.

Reply:
I see - good point.  When I started the refactoring, I basically did Ctrl+Shift+F looking for `SchemaModuleDict` and assuming all callers of functions with that as an arg should be amended - but you make a good point that the engine init logic _should_ use the dict versions. 

With later cleanup, it'd be good to encapsulate these types of helper functions inside the init/injection classes themselves so that their use is scoped to just engine init.